iterator.go

```diff
@@ -213,6 +217,8 @@ func (r RootLookuper) doIteratePass(ctx context.Context, req *dns.Msg,
 			return r.handleSuccessAnswer(ctx, req, resp, server)
 		case exdns.HasNsType(resp, dns.TypeNS):
 			return r.handleSuccessDelegation(ctx, req, resp, server)
+		case exdns.HasNsType(resp, dns.TypeSOA):
+			return r.handleSuccessAnswer(ctx, req, resp, server)
```

Comment:
this looks like abusing `handleSuccessAnswer`, see comment there.

Reply:
solved